crates/bitcoin-da/src/service.rs

```diff
@@ -338,322 +357,162 @@ impl BitcoinService {
-                )));
-            }
-
-            raw_txs.push(signed_raw_commit_tx.hex);
-
-            let serialized_reveal_tx = encode::serialize(&reveal);
-            raw_txs.push(serialized_reveal_tx);
-        }
-
-        let mut inputs = vec![];
-
-        for input in commit.input.iter() {
-            if let Some(entry) = all_tx_map.get(&input.previous_output.txid) {
-                inputs.push(SignRawTransactionInput {
-                    txid: input.previous_output.txid,
-                    vout: input.previous_output.vout,
-                    script_pub_key: entry.output[input.previous_output.vout as usize]
-                        .script_pubkey
-                        .clone(),
-                    redeem_script: None,
-                    amount: Some(entry.output[input.previous_output.vout as usize].value),
-                });
-            }
-        }
-        let signed_raw_commit_tx = self
-            .client
-            .sign_raw_transaction_with_wallet(&commit, Some(inputs.as_slice()), None)
+        tx_request: DaTxRequest,
+        fee_sat_per_vbyte: u64,
+    ) -> Result<()> {
+        let da_txs = self
+            .create_da_transactions_with_fee_rate(tx_request, fee_sat_per_vbyte)
             .await?;
+        let signed_txs = self.tx_signer.sign_da_txs(da_txs).await?;
+        self.test_mempool_accept_queue_tx(&signed_txs).await?;
 
-        if let Some(errors) = signed_raw_commit_tx.errors {
-            return Err(BitcoinServiceError::InvalidTransaction(format!(
-                "Failed to sign the aggregate commit transaction: {:?}",
-                errors
-            )));
-        }
-
-        raw_txs.push(signed_raw_commit_tx.hex);
-
-        let serialized_reveal_tx = encode::serialize(&reveal.tx);
-        raw_txs.push(serialized_reveal_tx);
+        // backup to file after mempool acceptance
+        backup_txs_to_file(&self.tx_backup_dir, &signed_txs)?;
 
-        self.test_mempool_accept(&raw_txs).await?;
+        self.queue_transactions(signed_txs).await;
+        Ok(())
+    }
 
-        // backup tx only if it passes mempool accept test
-        backup_chunked_txs(back_up_path, &raw_txs)?;
+    async fn queue_transactions(&self, txs: Vec<SignedTxPair>) {
+        self.tx_queue.lock().await.extend(txs);
+    }
 
-        // Track the sum of all chunked transactions sizes
-        let raw_txs_size_sum: usize = raw_txs.iter().map(|tx| tx.len()).sum();
-        histogram!("da_transaction_size").record(raw_txs_size_sum as f64);
+    /// Send transaction out of the queue to DA until the first error.
+    /// Returns the successfully sent txs.
+    pub(crate) async fn process_transaction_queue(&self) -> Result<Vec<Txid>> {
+        let mut queue = self.tx_queue.lock().await;
 
-        let txids = self.send_raw_transactions(&raw_txs).await?;
+        let mut txids = Vec::new();
+        while let Some(tx) = queue.front() {
+            if self.test_mempool_accept(&tx.as_raw_txs()).await.is_err() {
```

Comment:
So if we got an error in the middle of chunks, once sequencer receives another request (zkproof/seqcom), we will try to send it again and fail.

Instead we can do a queue [pack1, pack2, ...] where a pack is [bitcoin tx1, bitcoin tx2, ...]. And instead of picking the front elem, we pop the from elem. So at least we will be able to put seqcomms on DA if we stuck in the middle of chunk sending.